### PR TITLE
Kerman/deployment module

### PR DESCRIPTION
**Pull request by kermankohli**

### CODE REVIEW

File: artifacts/index.ts [hunk 1]
```diff
@@ -1,131 +0,0 @@
-import { BadTokenMock } from "./ts/BadTokenMock";
```
Comment (felix2feng) [inline artifacts/index.ts]: I think we need this file for exporting contracts in the `set-protocol-contracts` -package

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: 2
 jobs:
   checkout_and_compile:
     docker:
-      - image: circleci/node:9.6.1
+      - image: circleci/node:8.1.4
```
Comment (felix2feng) [inline .circleci/config.yml]: Any reason we're changing this config?

File: deployments/manager.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import { DeploymentStageInterface } from '../types/deployment_stage_interface';
+
+import { LibrariesStage } from './stages/1_libraries';
+import { CoreStage } from './stages/2_core';
+import { ModulesStage } from './stages/3_modules';
+import { RebalancingStage } from './stages/5_rebalancing';
+import { AuthorizationStage } from './stages/4_authorization';
+
+import { asyncForEach } from '../utils/array';
+import { getNetworkName, getNetworkId, returnOutputs, writeStateToOutputs, getContractAddress, removeNetwork, getContractCode } from './utils/output-helper';
```
Comment (felix2feng) [inline deployments/manager.ts]: Let's keep lines to 120 chars

File: deployments/network-constants.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import constants from "./constants";
+
+export default {
```
Comment (felix2feng) [inline deployments/network-constants.ts]: This is dope

File: deployments/outputs.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      "KyberNetworkWrapper": "0xd2aa8d362b1CaA68553642831b86Abb3D24B4579",
+      "ZeroExExchangeWrapper": "0x434f1EB003B78c0EAbe034313F1aFf47920e0860",
+      "LinearAuctionPriceCurve": "0x46c6A737C75cE3a58c6b2De14970E8841c72DcEF",
+      "ConstantAuctionPriceCurve": "0x1a488d7B42C1Ec1539b78f772BF13eCCB723f5fa",
+      "BitEthRebalanceManager": "0x57b7aB09008dD5eda262b160A7b0f0d17c8754B4",
+      "InitialCollateralSet": "0x22d34df714907847839addd687E2fd725d25d3a9",
+      "BitEthRebalancingSetToken": "0x0d751881E89BAA7e2238e9F9CEE54606EAf716b0"
+    },
+    "state": {
+      "last_deployment_stage": "5"
```
Comment (felix2feng) [inline deployments/outputs.json]: So do the deployments restart using the previous state AS well as taking into account already-deployed contracts?

File: deployments/stages/1_libraries.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    await this.deployStandardStartRebalanceLibrary();
+    await this.deployStandardPlaceBidLibrary();
+    await this.deployStandardFailAuctionLibrary();
+  }
+
+  private async deployERC20Wrapper(): Promise<ERC20WrapperContract> {
+    const name = 'ERC20Wrapper';
+    let address = await getContractAddress(name);
+
+    if (address) {
```
Comment (felix2feng) [inline deployments/stages/1_libraries.ts]: Perhaps there's a way to de-dupe this pattern?

File: deployments/outputs.json (hunk 5 above)
Comment (kermankohli) [reply deployments/outputs.json]: Yup.

**Pull request closed without merging**